### PR TITLE
Added binance signer

### PR DESCRIPTION
**Pull request by marktrs**

### CODE REVIEW

File: src/binance-signer.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+const { crypto } = require('@binance-chain/javascript-sdk')
+/**
+ * Signs message using a Binance account.
+ * This signer should be used for interactive signing in the browser with TronLink.
+ */
+export class BinanceSigner implements IEthereumSigner { 
+  private _privateKey: string
+  private _address: string
+
+  constructor(privateKey: string, accountAddress: string) {
```
Comment (enlight) [inline src/binance-signer.ts]: Shouldn't need to pass the address through, just get it from the private key.

File: src/binance-signer.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+export class BinanceSigner implements IEthereumSigner { 
+  private _privateKey: string
+  private _address: string
+
+  constructor(privateKey: string) {
+    this._privateKey = privateKey
+    const address = crypto.getAddressFromPrivateKey(privateKey)
+    const decod = crypto.decodeAddress(address)
+    const binanceAddressHex = "0x" + decod.toString("hex")
+    this._address = binanceAddressHex
```
Comment (enlight) [inline src/binance-signer.ts]:
```typescript
this._address = "0x" + crypto.decodeAddress(address).toString("hex")
```

File: src/binance-signer.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * Signs message using a Binance account.
+ * This signer should be used for interactive signing in the browser.
+ */
+export class BinanceSigner implements IEthereumSigner { 
+  private _privateKey: string
+  private _address: string
+
+  constructor(privateKey: string) {
+    this._privateKey = privateKey
```
Comment (enlight) [inline src/binance-signer.ts]: Convert the key to a `Buffer` here, instead of doing it every time it's used to sign.

File: tsconfig.json [hunk 4]
```diff
@@ -3,7 +3,7 @@
     /* Basic Options */
     "target": "es5",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
     "module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
-    "lib": ["es2015", "dom"],                /* Specify library files to be included in the compilation. */
+    "lib": ["es2015", "dom", "es2017.string"],                /* Specify library files to be included in the compilation. */
```
Comment (enlight) [inline tsconfig.json]: Why did this change?

File: src/binance-signer.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import { bytesToHex } from './crypto-utils';
+import { IEthereumSigner } from './solidity-helpers'
+import ethutil from 'ethereumjs-util'
+const { crypto } = require('@binance-chain/javascript-sdk')
+/**
+ * Signs message using a Binance account.
+ * This signer should be used for interactive signing in the browser.
+ */
+export class BinanceSigner implements IEthereumSigner {
```
Comment (enlight) [inline src/binance-signer.ts]: We should probably rename `IEthereumSigner` to `IMessageSigner`, but that can go in another PR.

File: tsconfig.json (hunk 4 above)
Comment (marktrs) [reply tsconfig.json]: removed, it was imported to allow using es7 string features

File: src/contracts/address-mapper.ts [hunk 6]
```diff
@@ -54,6 +54,7 @@ export class AddressMapper extends Contract {
     return this.callAsync<void>('AddIdentityMapping', mappingIdentityRequest)
   }
 
+  //Similar to addIdentityMappingAsync but for Binance, and this function not require message encryption
```
Comment (enlight) [inline src/contracts/address-mapper.ts]: That's not true, it doesn't require message hashing since that's presumably done by the BinanceSigner. Also see the other method comments, they use `/** */` comments, use that.

File: src/crypto-utils.ts [hunk 7]
```diff
@@ -34,7 +34,13 @@ export function getGUID(): string {
...
 export const PRIVATE_KEY_LENGTH = nacl.sign.secretKeyLength
 export const PUBLIC_KEY_LENGTH = nacl.sign.publicKeyLength
-
+export const enum SIGNATURE_TYPE {
+  EIP712  = 0,
+	GETH    = 1,
+	TREZOR  = 2,
+	TRON    = 3,
+	BINANCE = 4
+}
```
Comment (enlight) [inline src/crypto-utils.ts]: Formatting is messed up, run `yarn format`

File: yarn.lock [hunk 8]
```diff
@@ -7732,14 +7069,12 @@ yargs@^12.0.4:
...
   version "2.4.1"
   resolved "https://registry.yarnpkg.com/yauzl/-/yauzl-2.4.1.tgz#9528f442dab1b2284e58b4379bb194e22e0c4005"
-  integrity sha1-lSj0QtqxsihOWLQ3m7GU4i4MQAU=
   dependencies:
     fd-slicer "~1.0.1"
 
 yauzl@^2.4.2:
   version "2.10.0"
   resolved "https://registry.yarnpkg.com/yauzl/-/yauzl-2.10.0.tgz#c7eb17c93e112cb1086fa6d8e51fb0667b79a5f9"
-  integrity sha1-x+sXyT4RLLEIb6bY5R+wZnt5pfk=
```
Comment (enlight) [inline yarn.lock]: What version of yarn are you using?
